### ISSUE TITLE
Suggestion: A few changes

### ISSUE THREAD

**Issue by colombeen** [dnewsholme/PasswordState-Management#52]
Delete Get-PasswordStatePasswords and make it a part of Find-PasswordStatePassword
Rename Find-PasswordStatePassword to Get-PasswordStatePassword

Show a list of all passwords when no params are provided or when just a passwordlistid was passed on

Add aliases:
```powershell
PS C:\> New-Alias -Name Find-PasswordStatePassword -Value Get-PasswordStatePassword -Force
PS C:\> New-Alias -Name Get-PasswordStatePasswords -Value Get-PasswordStatePassword -Force
```

I'm just looking at this in comparison to "Get-Child" or something similar. Just seems more logical to provide everything if you don't filter, and filter with the provided params if neccesary.

**Comment by dnewsholme** [comment]
I'm happy to alias Find-PasswordStatePassword but think that the Get-PasswordStatePasswords alias is probably unnecessary. Think we should probably use the opportunity to drop the plural from the cmdlet name since it doesn't conform to the best practices for naming the cmdlet.

This would make it a breaking change but it's not huge.

With regards to returning all by default with no parameters specified i think it would simplify the module and worth doing to.

So to summarize the scope of work would be: 

1. Add the following Alias.

```powershell
New-Alias -Name Find-PasswordStatePassword -Value Get-PasswordStatePassword -Force
```
2. Rename Find-PasswordstatePassword to Get-PasswordStatePassword

3. Make Get-PasswordStatePassword return all passwords by default with no parameters specified.

4. Delete function Get-PasswordStatePasswords.

**Comment by colombeen** [comment]
Sounds good

**Comment by dnewsholme** [comment]
Made the changes in #53. I've checked and i don't think adding the wildcard broke anything, i've tested a few scenarios and it seemed to work.

**Comment by colombeen** [comment]
I've tried using the API with the % to retrieve them all... It works but not in combination with the PasswordListID

(Get-PasswordStateResource -uri '/api/searchpasswords/?search=%).count
(Get-PasswordStateResource -uri '/api/searchpasswords/?search=%&passwordlistid=734').count
(Get-PasswordStateResource -uri '/api/searchpasswords/?search=%25&passwordlistid=734').count
all show me 5418 instead of 18 for list 734

**Comment by dnewsholme** [comment]
> I've tried using the API with the % to retrieve them all... It works but not in combination with the PasswordListID
> 
> ```powershell
> (Get-PasswordStateResource -uri '/api/searchpasswords/?search=%').count
> (Get-PasswordStateResource -uri '/api/searchpasswords/?search=%&passwordlistid=734').count
> (Get-PasswordStateResource -uri '/api/searchpasswords/?search=%25&passwordlistid=734').count
> ```
> 
> all show me 5418 instead of 18 for list 734

When you search as passwordlist you do 
```powershell
(Get-PasswordStateResource -uri '/api/searchpasswords/734?search=%25).count
```
Which is what Get-PasswordStatePassword does.

```powershell
C:\git\PasswordState-Management [dev ≡]> (Get-PasswordStatePassword -Verbose).count
VERBOSE: using uri https://pws.linkgroup.co.uk/winapi/searchpasswords/?Search=%25
VERBOSE: GET https://pws.linkgroup.co.uk/winapi/searchpasswords/?Search=%25 with 0-byte payload
VERBOSE: received 227550-byte response of content type application/json; charset=utf-8
429
C:\git\PasswordState-Management [dev ≡]> (Get-PasswordStatePassword -PasswordListID 24 -Verbose).count
VERBOSE: using uri https://pws.linkgroup.co.uk/winapi/searchpasswords/24?Search=%25
VERBOSE: GET https://pws.linkgroup.co.uk/winapi/searchpasswords/24?Search=%25 with 0-byte payload
VERBOSE: received 7407-byte response of content type application/json; charset=utf-8
15
```

**Comment by colombeen** [comment]
I would try something like this : 

```powershell
# Line 132
Switch ($PSCmdlet.ParameterSetName) {
  # General search
  'General' {
    If ([string]::IsNullOrEmpty($Search)) {
      $uri += "/api/passwords/$($PasswordlistID)?QueryAll"
    }
    Else {
      $uri += "/api/searchpasswords/$($PasswordListID)?Search=$([System.Web.HttpUtility]::UrlEncode($Search))"
    }
  }
  # Search on a specific password ID
```

**Comment by colombeen** [comment]
Oh my bad I forgot that the PwdListID needs to go before the ?

**Comment by dnewsholme** [comment]
But using the queryall parameter on the url string is probably safer than passing a SQL wildcard character as the behaviour could change between versions. So i think i'll make the change how you suggest anyway.

**Comment by colombeen** [comment]
We could also remove the + from $uri += because the variable is never appended with anything

**Comment by dnewsholme** [comment]
I'll do that too.

I found a weird quirk in the API when using queryall.
```powershell
C:\git\PasswordState-Management [dev ≡ +0 ~2 -0 !]> (Get-PasswordStatePassword -Verbose).TreePath
VERBOSE: using uri https://pws.linkgroup.co.uk/winapi/passwords/?QueryAll
VERBOSE: GET https://pws.linkgroup.co.uk/winapi/passwords/?QueryAll with 0-byte payload
VERBOSE: received 248937-byte response of content type application/json; charset=utf-8
\FS DevOps\FS DevOps Systems\Production
```
It returns an extra property called TreePath on the object, which caused it to not cast to the PasswordResult class.

When doing any other search the property isn't there.

```powershell
C:\git\PasswordState-ManagementPS> (Get-PasswordStatePassword test -Verbose).TreePath
VERBOSE: using uri https://pws.linkgroup.co.uk/winapi/searchpasswords/?Search=test
VERBOSE: GET https://pws.linkgroup.co.uk/winapi/searchpasswords/?Search=test with 0-byte payload
VERBOSE: received 12575-byte response of content type application/json; charset=utf-8
```
searching all within a passwordlist doesn't do this either.

```powershell
C:\git\PasswordState-Management [dev ≡ +0 ~2 -0 !]> (Get-PasswordStatePassword -PasswordListID 24 -Verbose).TreePath
VERBOSE: using uri https://pws.linkgroup.co.uk/winapi/passwords/24?QueryAll
VERBOSE: GET https://pws.linkgroup.co.uk/winapi/passwords/24?QueryAll with 0-byte payload
VERBOSE: received 6627-byte response of content type application/json; charset=utf-8
```

**Comment by colombeen** [comment]
Strange

**Comment by dnewsholme** [comment]
Merged into release 3.0.0
